.github/workflows/build-on-feature-branch.yaml

```diff
@@ -5,16 +5,22 @@ on:
       - main
 
 jobs:
-  release:
-    name: Build Docker
-    uses: timescale/cloud-actions/.github/workflows/build.yaml@main
-    with:
-      tags: |
-        auto-${{ github.sha }}
-      registry: 142548018081.dkr.ecr.us-east-1.amazonaws.com/tiger-docs-mcp-server
-      dockerfile_path: ./Dockerfile
-      skip-scan: true
-    secrets:
-      API_TOKEN_GITHUB: ${{ secrets.API_TOKEN_GITHUB }}
-      AWS_ACCESS_KEY_ID: ${{ secrets.ORG_AWS_ACCESS_KEY_ID }}
-      AWS_SECRET_ACCESS_KEY: ${{ secrets.ORG_AWS_SECRET_ACCESS_KEY }}
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Dispatch build-on-feature-branch workflow
+        uses: actions/github-script@v8
+        with:
+          github-token: ${{ secrets.GH_AGENTS_PAT }}
+          script: |
+            await github.rest.actions.createWorkflowDispatch({
+              owner: 'timescale',
+              repo: 'tiger-agents-deploy',
+              workflow_id: 'build-on-feature-branch.yaml',
+              ref: 'main',
+              inputs: {
+                repository: 'tiger-docs-mcp-server',
+                ref: '${{ github.ref }}',
```

Comment:
I'm a little worried about the security of this. Could someone make a PR that modifies this workflow in such a way that it triggers deploying their (malicious) branch to production?

It might be safer to say that it only works with `main`, doesn't take a ref.

Reply:
The attack vector would require the attacker to have commit access to this repository as this flow only runs on `push` and requires the `secret.GH_AGENTS_PAT` token.

Like yeah, at that point they could load malicious code, but they could then also probably do a similar thing just pushing to `main` as well.